packages/tooltips/src/elements/Tooltip.tsx

```diff
@@ -14,46 +14,9 @@ import { useTooltip } from '@zendeskgarden/container-tooltip';
 import { composeEventHandlers, getControlledValue } from '@zendeskgarden/container-utilities';
 import { Manager, Popper, Reference } from 'react-popper';
 import { Modifiers } from 'popper.js';
-
-import {
-  getPopperPlacement,
-  getRtlPopperPlacement,
-  GARDEN_PLACEMENT
-} from '../utils/gardenPlacements';
-import { StyledTooltipWrapper, StyledTooltip, TOOLTIP_SIZE, TOOLTIP_TYPE } from '../styled';
-
-export interface ITooltipProps extends HTMLAttributes<HTMLDivElement> {
-  /** Appends the tooltip to the element provided */
-  appendToNode?: Element;
-  /** Adds an arrow to the tooltip */
-  hasArrow?: boolean;
-  /** Adds milliseconds of delay to the opening and closing of the tooltip */
-  delayMS?: number;
-  /** Allows the tooltip to reposition during browser resize events */
-  eventsEnabled?: boolean;
-  /** Defines the content of the tooltip */
-  content: React.ReactNode;
-  /**
-   * Adjusts the placement of the tooltip
-   **/
-  placement?: GARDEN_PLACEMENT;
-  /** Passes configurations to the [Popper instance](https://popper.js.org/docs/v2/modifiers/) */
-  popperModifiers?: Modifiers;
-  /** Adjusts the padding and font size */
-  size?: TOOLTIP_SIZE;
-  /** Specifies the tooltip type */
-  type?: TOOLTIP_TYPE;
-  /** Sets the `z-index` of the tooltip */
-  zIndex?: number | string;
-  /** Displays the tooltip on initial render */
-  isInitialVisible?: boolean;
-  /** Displays the tooltip */
-  isVisible?: boolean;
-  /** @ignore ReactNode override */
-  children: React.ReactElement;
-  /** Defines the ref key used to position the tooltip */
-  refKey?: string;
-}
+import { getPopperPlacement, getRtlPopperPlacement } from '../utils/gardenPlacements';
+import { StyledTooltipWrapper, StyledTooltip } from '../styled';
+import { ITooltipProps, PLACEMENT, SIZE, TYPE } from '../utils/types';
```

Comment:
In this PR's approach, the public interface pushes _out_ rather than pulling the interface _up_ from styled view elements. I believe this is the better angle, as the public interface is focal for typed prop data and documentation. And any styled interfaces should pull from the public interface as-needed.

Reply:
For example, with the public interface pushed _out_, a styled component's interface could look something like the following without creating a circular code reference:

```
import { ITooltipProps } from '...';

interface IStyledTooltipProps {
  reusedProp: ITooltipProps['prop'];
}
```

Reply:
The component author will need to take care with this ☝️ type of reuse, as `undefined` can come through from the imported interface in a manner that makes the local styled interface feel less expressive. In this case, it could appear that `reusedProp` is mandatory. But if `ITooltipProps['prop']` is optional, then `reusedProp` may also be `undefined`.

tldr; I'm not entirely convinced this is an ideal reuse pattern.

Reply:
fwiw, there are [TypeScript utilities](https://www.typescriptlang.org/docs/handbook/utility-types.html) to help with this. It is just a matter of the author being meticulous. For example:

```ts
import { ITooltipProps } from '...';

interface IStyledTooltipProps {
  reusedProp: NonNullable<ITooltipProps['prop']>;
}
```